src/FuncSharp/Numeric/Digit.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+
+namespace FuncSharp;
+
+public readonly struct Digit
+{
+    private readonly int value;
```

Comment:
can this be of a `byte` type? As it holds a single digit

Reply:
Yup, following PRs with tests :)